File: 2021/go/day01/second/second.go
package main

import (
	"bufio"
	"log"
	"os"
	"strconv"
)

func main() {
	counter := 0

	var depths []int

	file, err := os.Open("../input.txt")

	if err != nil {
		log.Fatalf("Can not open file")

	}

	defer file.Close()

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		deep, _ := strconv.Atoi(scanner.Text())
		depths = append(depths, deep)
		l := len(depths)
		if l > 3 {
			if depths[l-1] > depths[l-4] {
				counter++
			}
		}
	}

	log.Println(counter)
}

File: 2015/day01/second.py
f = open("input.txt")
str = f.read()

i = 0
count = 0
for c in str:
	i+=1
	if c == '(':
		count+=1
	if c == ')':
		count-=1

	if count == -1:
		print(i)
		break		